ui/src/routes/conversations/[id]/ConversationMessages.svelte

```diff
@@ -39,36 +136,52 @@
 </script>
 
 <div class="flex w-full flex-1 flex-col-reverse p-4">
-  <ul>
-    {#each messages as [actionHashB64, messageExtended], i (actionHashB64)}
-      {@const prevMessageExtended = i === 0 ? undefined : messages[i - 1][1]}
-
-      {#if prevMessageExtended === undefined || !isSameDay(new Date(messageExtended.timestamp / 1000), new Date(prevMessageExtended.timestamp / 1000))}
-        <li class="my-4">
-          <div class="text-secondary-400 dark:text-secondary-300 text-center text-xs">
-            {new Date(messageExtended.timestamp / 1000).toLocaleDateString("en-US", {
-              weekday: "long",
-              month: "long",
-              day: "numeric",
-            })}
-          </div>
-        </li>
-      {/if}
-
-      <BaseMessage
-        {cellIdB64}
-        message={messageExtended}
-        isSelected={selected === actionHashB64}
-        showAuthor={prevMessageExtended === undefined ||
-          messageExtended.authorAgentPubKeyB64 !== prevMessageExtended.authorAgentPubKeyB64 ||
-          !isWithinFiveMinutes(
+  <div class="scroll-container" bind:this={virtualListEl} on:scroll={handleScroll}>
+    <div style="position: relative; height: {$virtualizer.getTotalSize()}px; width: 100%;">
+      {#each $virtualizer.getVirtualItems() as virtualItem (messages[virtualItem.index][0])}
+        {@const [actionHashB64, messageExtended] = messages[virtualItem.index]}
+        {@const prevMessage =
```

Comment:
```suggestion
        {@const prevMessageExtended =
```

The rest of the codebase has been very consistent in variable naming to make clear when we're using *Extended types. Let's keep it that way.